bpf/lib/nodeport.h

```diff
@@ -1928,13 +1931,14 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 			return DROP_NOT_IN_SRC_RANGE;
 #if defined(ENABLE_L7_LB)
 		if (lb4_svc_is_l7loadbalancer(svc) && svc->l7_lb_proxy_port > 0) {
-#if __ctx_is == __ctx_xdp
-			/* We cannot redirect from the XDP layer to cilium_host.
-			 * Therefore, let the bpf_host to handle the L7 ingress
-			 * request.
-			 */
-			return CTX_ACT_OK;
-#endif
+			if (ctx_is_xdp()) {
+				/* We cannot redirect from the XDP layer to cilium_host.
+				 * Therefore, let the bpf_host to handle the L7 ingress
+				 * request.
+				 */
+				return CTX_ACT_OK;
+			}
```

Comment:
nit: you could move the comment above the if (ctx_is_xdp()), and then just
if (ctx_is_xdp())
            return CTX_ACT_OK;
without the braces.